src/_guides/language/extension-methods.md

```diff
@@ -0,0 +1,246 @@
+---
+title: Extension methods
+description: Learn how to add to existing APIs.
+---
+<?code-excerpt replace="/ *\/\/\s+ignore_for_file:[^\n]+\n//g; /([A-Z]\w*)\d\b/$1/g; /\b(main)\d\b/$1/g"?>
+
+Extension methods, introduced in Dart 2.7,
+are a way to add functionality to existing libraries.
+You might not ever need to _use_ extension methods, much less _create_ them.
+But if you generate code or find yourself writing lots of wrappers,
+consider using extension methods to reduce
+the amount of boilerplate code that you need to write.
+
+You might use extension methods without even knowing it.
+For example, when you use code completion in an IDE,
+it suggests extension methods alongside regular methods.
```

Comment:
I wonder if we rank extension methods differently from other methods. Maybe check with the analyzer team?

Reply:
I hesitate to get too specific about exactly where the extension methods are displayed.